glean-core/csharp/Glean/Metrics/QuantityMetricType.cs

```diff
@@ -0,0 +1,142 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+using Mozilla.Glean.FFI;
+using System;
+
+namespace Mozilla.Glean.Private
+{
+    /// <summary>
+    /// This implements the developer facing API for recording quantity metrics.
+    /// 
+    /// Instances of this class type are automatically generated by the parsers at build time,
+    /// allowing developers to record values that were previously registered in the metrics.yaml file.
+    ///
+    /// The quantity API only exposes the [set] method.
+    //
+    /// </summary>
+    public sealed class QuantityMetricType {
+        private bool disabled;
+        private string[] sendInPings;
+        private UInt64 handle;
+
+        /// <summary>
+        /// The public constructor used by automatically generated metrics.
+        /// </summary>
+        public QuantityMetricType(
+            bool disabled,
+            string category,
+            Lifetime lifetime,
+            string name,
+            string[] sendInPings
+            ) : this(0, disabled, sendInPings)
+        {
+            handle = LibGleanFFI.glean_new_quantity_metric(
+                    category: category,
+                    name: name,
+                    send_in_pings: sendInPings,
+                    send_in_pings_len: sendInPings.Length,
+                    lifetime: (int)lifetime,
+                    disabled: disabled);
+        }
+
+        internal QuantityMetricType(
+            UInt64 handle,
+            bool disabled,
+            string[] sendInPings
+            )
+        {
+            this.disabled = disabled;
+            this.sendInPings = sendInPings;
+            this.handle = handle;
+        }
+
+
+        /// <summary>
+        /// Set a quantity value.
+        /// </summary>
+        /// <param name="value">The value to set. Must be non-negative.</param>
+        public void Set(Int32 value)
+        {
+            if (disabled)
+            {
+                return;
+            }
+
+            Dispatchers.LaunchAPI(() => {
+                SetSync(value);
+            });
+        }
+
+        /// <summary>
+        /// Internal only, synchronous API for setting a quantity value.
+        /// </summary>
+        /// <param name="value">The value to set. Must be non-negative.</param>
+        internal void SetSync(Int32 value)
```

Comment:
Do we need this? We're not really using it anywhere.

Reply:
We use it in line 68, if you mean the `SetSync`. Or do you mean the whole C# part?

Reply:
Just saw your comment about not droppping the C# part.

Reply:
Yes, but we just use it there. We have the "sync" functions to make it possibly to synchronously set metrics during the Glean  init. But, for this case, we don't need it. Let's move the content of this function to the only call site, please! :)